### DISCUSSION TITLE
@cache decorator causes function to be untyped

### DISCUSSION THREAD

**Discussion by veber-alex** [microsoft/pyright#8466] [General]
```python
from functools import cache


@cache
def foo(_: int):
    """
    foo
    """
    pass


foo("test")

```
This doesn't show any error for me in vscode and even hovering over the call of foo just shows `(function) foo: _lru_cache_wrapper[None]`

I am not sure if the issue is in pyright, vscode or typeshed.

**Comment by erictraut** [comment]
This is a [known issue](https://github.com/python/typeshed/issues/11280) with the typeshed definition of `functools.cache`. 

As discussed in that thread, there are tradeoffs that the typeshed maintainers are trying to balance here. I'm not convinced they've struck the right balance in this case. Feel free to add to the typeshed thread if you have an opinion.
